src/alert.cpp

```diff
@@ -250,8 +251,8 @@ bool CAlert::ProcessAlert(bool fThread)
                 // the whole string before passing it to the shell:
                 std::string singleQuote("'");
                 // safeChars chosen to allow simple messages/URLs/email addresses, but avoid anything
-                // even possibly remotely dangerous like & or >
-                std::string safeChars("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ01234567890 .,;_/:?@");
+                // even possibly remotely dangerous like & ; $ or >
+                std::string safeChars("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ01234567890 .,_/:?@");
```

Comment:
Is this stored in the appcache later? This should be much safer with the appcache map-map changes, right?

Reply:
No, the safened string is passed as a command-line argument to a user specified command. I removed a semicolon character from the set of safe characters as it is often used for separating commands. The string is enclosed in single quotes later, so yeah, the semicolon can't do much there. Extra caution does not hurt I guess.